src/components/App.js

```diff
@@ -2,25 +2,43 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { fetchAllMovies } from '../apiCalls';
 import { addMovies } from '../actions/index';
+import { Route } from 'react-router-dom';
 import NavigationBar from '../containers/NavigationBar';
-
+import MovieContainer from '../containers/MovieContainer';
+import MovieShowPage from '../components/MovieShowPage';
 
 export class App extends Component {
   componentDidMount() {
-    fetchAllMovies().then(data => this.props.addMovies(data.movies));
+    const { addMovies } = this.props;
+    fetchAllMovies().then(data => addMovies(data.movies));
   }
 
-  render() {
+  render = () => {
     return (
       <main>
         <NavigationBar />
+
+        <Route exact path='/' render={props => <MovieContainer {...props} />} />
+
+        <Route
+          exact
+          path='/movie/:id'
+          render={({ match }) => {
+            let moviesData = [...this.props.movies];
+            return <MovieShowPage {...moviesData} />;
+          }}
+        />
```

Comment:
I will need to review notes on Router to better understand using {match} here.  It doesn't look like we are using it yet but will need to render based on the movie id.

I also will need to review my notes about what is being passed through as props on line 28.  I'm a little confused about how its being passed through (declared as a variable, passed not as a key/value pair) and whether we need to pass it this way or if we can just get that info within the component from the store.

I'm just fuzzy on these two things - it doesn't mean they are incorrect or need to be changed.   :)